tutorials/arvo/ames.md

```diff
@@ -112,40 +115,299 @@ ducts as unique keys.  Stop tracking a peer by sending Ames a
+contact `~worwel-sipnum` but does not know the IP address and port at which
+`~worwel-sipnum` resides. Both planets live under `~zod`, which knows both of
+their IP and port numbers by virtue of being the galaxy that both live under.
+
+So `~bacbel-tagfeb` sends a packet addressed to `~worwel-sipnum` to `~zod`.
+
+The packet has the following format:
+ - The standard Ames header described in [header](#header), where the checksum
+   is the `+mug` of the body, and the sender and receiver ship types are `01`,
+   denoting that the sender and receiver are planets. The remainder of the
+   packet is the body.
+ - The body of this packet will be the `@p` of the
+sender `~bacbel-tagfeb`, followed by the `@p` of the receiver `~worwel-sipnum`,
+followed by the payload.
+ - The payload of this packet will be the `+jam` of `[origin content]`. `origin`
+for this initial packet will be `~`, which implies that the packet is
+originating from the sender. `content` is the `+jam` of the message, encrypted
+using a Diffie-Hellman symmetric key obtained with the private key of
+`~bacbel-tagfeb` and the public key of `~worwel-sipnum`.
+
+`~zod` receives the packet and reads the body. It sees that it is not the
+intended recipient of the packet, and so gets ready to forward it to
+`~worwel-sipnum`. First, it [`+cue`s](@/docs/reference/library/2p.md#cue)
+(deserializes) the payload and changes the `origin` to the IP and port of
+`~bacbel-tagfeb`. Then it `+jam`s `[origin content]` to form a new payload, and
+`+mug`s that payload to get a new checksum. It replaces the old payload and
+checksum with the new, and then sends the packet along to `~worwel-sipnum`.
+
+Once `~worwel-sipnum` processes the packet, it will know the IP and port of
+`~bacbel-tagfeb` since `~zod` included it when it forwarded the packet and so
+will send an ack packet directly to `~bacbel-tagfeb`. Communication between the
+two ships will now be direct until one of them changes their IP address, port, or networking keys.
+
+
+## The Serf and the King
+
+Urbit's functionality is split between the two binaries `urbit-worker` (sometimes
+called the Serf) and `urbit-king` (sometimes called the King). This division of
+labor is currently not well-documented outside of the [Vere documents](@/docs/tutorials/vere/_index.md), but we summarize it here.
+
+In short, the Serf is the Nock runtime and so keeps track of the current state
+of Arvo as a Nock noun and updates the state by `%poke`ing it with nouns, and
+then informs the King of the new state. The King manages snapshots of the Arvo
+state and handles I/O with Unix, among other things. The Serf only ever talks to
+the King, while the King talks with both the Serf and Unix.
+
+### Ames
+
+The King has several submodules, one of them being an Ames I/O submodule. This
+submodule is responsible for wrapping outgoing Ames packets as UDP packets, and
+looking at incoming UDP packets handed to it by Unix to ensure that they are
```

Comment:
It doesn't do much validation; it mostly turns the packets into Ames packets and sends them to the worker to be processed.